rero_ils/modules/acq_orders/api.py

```diff
@@ -200,3 +203,15 @@ def bulk_index(self, record_id_iterator):
         :param record_id_iterator: Iterator yielding record UUIDs.
         """
         super().bulk_index(record_id_iterator, doc_type='acor')
+
+    def delete(self, record):
+        """Delete a record from indexer.
+
+        First delete order lines from the ES index, then delete the order.
+        """
+        es_query = AcqOrderLinesSearch()\
+            .filter('term', acq_order__pid=record.pid)
+        if es_query.count():
+            es_query.delete()
+            AcqOrderLinesSearch.flush_and_refresh()
```

Comment:
Could you put everything into the `pre_delete` function. Seems confused to have multiple codes on different places.

Reply:
as discussed, we separate the deletion from DB and ES.